tests/AppBundle/Controller/NewsletterControllerTest.php

```diff
@@ -92,6 +94,43 @@ public function testSubscriptionFromHome()
         $this->assertCount(1, $this->getMailjetEmailRepository()->findMessages(NewsletterSubscriptionMessage::class));
     }
 
+    /**
+     * @group functionnal
+     */
+    public function testInvitationAndRetry()
+    {
+        $this->assertCount(5, $this->subscriptionsRepository->findAll());
+
+        // Initial form
+        $crawler = $this->client->request(Request::METHOD_GET, '/newsletter/invitation');
+
+        $this->assertResponseStatusCode(Response::HTTP_OK, $this->client->getResponse());
+
+        $this->client->submit($crawler->filter('form[name=newsletter_invitation]')->form([
+            'newsletter_invitation[firstName]' => 'Titouan',
+            'newsletter_invitation[lastName]' => 'Galopin',
+            'newsletter_invitation[guests][0]' => 'hugo.hamon@sensiolabs.com',
+            'newsletter_invitation[guests][1]' => 'jules.pietri@sensiolabs.com',
+        ]));
+
+        $this->assertResponseStatusCode(Response::HTTP_FOUND, $this->client->getResponse());
+        $this->assertClientIsRedirectedTo('/newsletter/invitation/merci', $this->client);
+
+        // Invitations should have been saved
+        $this->assertCount(2, $invitations = $this->manager->getRepository(NewsletterInvite::class)->findAll());
+
+        /** @var NewsletterInvite $invite */
+        $invite = $invitations[0];
+
+        // Email should have been sent
+        $this->assertCount(2, $messages = $this->getMailjetEmailRepository()->findMessages(NewsletterInvitationMessage::class));
+
+        $this->assertContains('/newsletter?mail=hugo.hamon@seniolabs.com', $messages[0]->getRequestPayloadJson());
```

Comment:
The typo was to make the test fail this was still WIP because of the failure I didn't understand.